Automattic/wp-calypso
# Stats: Put 30-days summary behind paywall

Related to: https://github.com/Automattic/red-team/issues/317

## Proposed Changes

Adds the "30 days" button in the Summary page navigation to the list of paywalled features. This means the button will have the lock icon and clicking it will throw up the upsell card instead of updating the routing.

<img width="736" alt="SCR-20241212-ojqd" src="https://github.com/user-attachments/assets/74bbc36f-b3ac-4fb8-9bd3-e53f2b62e30f" />

## Why are these changes being made?


For consistency with other locked ranges.

## Testing Instructions



- Open the Calypso Live site.
- Navigate to a new site without a paid plan.
- Manually update the URL to a summary page: `/stats/day/referrers/SITE_SLUG`
- Confirm the "30 Days" button is locked.

Will be gated for all sites after 2024-01-09 without a paid plan.

## Pre-merge Checklist
